src/scene_map.cpp

```diff
@@ -392,6 +396,58 @@ void Scene_Map::OnAsyncSuspend(F&& f, AsyncOp aop, bool is_preupdate) {
 		screen_erased_by_event = false;
 	}
 
+	if (aop.GetType() == AsyncOp::eCallInn) {
+		activate_inn = true;
+		music_before_inn = Game_System::GetCurrentBGM();
+		inn_continuation = std::forward<F>(f);
+
+		Game_System::BgmFade(800);
+
+		// FIXME: Is 36 correct here?
+		Graphics::GetTransition().Init(Transition::TransitionFadeOut, Scene::instance.get(), 36, true);
+
+		AsyncNext([=]() { StartInn(); });
+		return;
+	}
+
 	AsyncNext(std::forward<F>(f));
 }
 
+void Scene_Map::StartInn() {
+	const RPG::Music& bgm_inn = Game_System::GetSystemBGM(Game_System::BGM_Inn);
+	if (bgm_inn.name.empty() ||
+		bgm_inn.name == "(OFF)" ||
+		bgm_inn.name == "(Brak)") {
+		FinishInn();
+		return;
+	}
+
+	Game_System::BgmStop();
```

Comment:
It's not necessary to call BgmStop before BgmPlay

Reply:
Fixed